File: Binary Tree Right Side View/binary-tree-right-side-view.java
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if (root == null) return res;

        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);

        while (!q.isEmpty()) {
            int levelSize = q.size();
            TreeNode lastNode = null;

            for (int i = 0; i < levelSize; i++) {
                TreeNode currentNode = q.poll();
                lastNode = currentNode;

                if (currentNode.left != null) q.add(currentNode.left);
                if (currentNode.right != null) q.add(currentNode.right);
            }

            res.add(lastNode.val);
        }

        return res;
    }
}

File: Binary Tree Right Side View/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        res = []
        if not root:
            return res
        
        from collections import deque
        q = deque([root])
        
        while q:
            level_size = len(q)
            last_node = None
            
            for _ in range(level_size):
                current_node = q.popleft()
                last_node = current_node
                
                if current_node.left:
                    q.append(current_node.left)
                if current_node.right:
                    q.append(current_node.right)
            
            res.append(last_node.val)
        
        return res
